### PR TITLE
[fat] Fix everything required for ELKS to boot on FAT

### PR DESCRIPTION
**Pull request by ghaerr**
fix msdos_file_read for read count > 32k
	always install ash as /bin/sh (no more /bin/ash)
	add /dev/eth (9,0) and /dev/tcpdev (8,0) to FAT /dev for ktcp support (untested)
	rename /etc/rc.d/rc.sysinit to /etc/rc.d/rc.sys in /etc/inittab for FAT compatibility
	add ELKS build commit ID to /etc/motd for tracking tester feedback
	display detailed FAT disk details on mount for tester feedback
	remove exec /etc/init, /sbin/init from init_task, use only /bin/init
	cleanup /bin/init error messages
	add temporary setboot to package manager FAT image builds to allow .config disk boot
	setboot needs updating to write ELKS -B parms but not clobber FAT BPB

ELKS now supports booting from MSDOS-formatted disks. I think it works quite well :)

No known bugs.

Do 'cd images; make images' after build, then use fd1440-fat.bin for boot disk.
Only 1440k disks supported until setboot is updated.
Also tested with seperately-mounted 360k FAT disk on /dev/fd1 with ELKS running on booted FAT filesystem: ELKS runs fine with no MINIX filesystems required!

Have fun!

### CODE REVIEW

File: elkscmd/rootfs_template/var/www/index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<BODY bgcolor="#ffffff">
+<CENTER>
+<B><BIG>It worked!!</BIG></B>
+<P>
+If you can see this page the web server worked.<br>
+Visit <a href="http://elks.sourceforge.net/">http://elks.sourceforge.net/</a>
+</p>
+</CENTER>
+</BODY>
+</HTML>
```
Comment (mfld-fr) [inline elkscmd/rootfs_template/var/www/index.html]: Duplicated page from 'httpd'. To be installed only when 'httpd' is selected in the configuration or in the package manager.

File: elkscmd/sys_utils/init.c [hunk 2]
```diff
@@ -81,14 +83,16 @@ int hash(const char *string)
...
-#endif
+/* Print an error message and die*/
+static void fatalmsg(const char *msg, ...)
+{
+	va_list args;
+	va_start(args, msg);
+	fprintf(stderr, "%s: ", initname);
+	vfprintf(stderr, msg, args);
+	va_end(args);
+	exit(-1);
```
Comment (mfld-fr) [inline elkscmd/sys_utils/init.c]: Dont't change the exit code semantic. Keep 1 as exit code on failure.

File: elkscmd/ash/Makefile [hunk 3]
```diff
@@ -46,8 +46,9 @@ ash: $(OBJS)
 
 install: ash
 ifdef CONFIG_IMG_LINK
-	$(INSTALL) ash $(DESTDIR)/bin
-	$(LN) -sf ash $(DESTDIR)/bin/sh
+#	$(INSTALL) ash $(DESTDIR)/bin
+#	$(LN) -sf ash $(DESTDIR)/bin/sh
+	$(INSTALL) ash $(DESTDIR)/bin/sh
```
Comment (mfld-fr) [inline elkscmd/ash/Makefile:48]: Useless 'ifdef' after the change. To be simplified.

**Review by mfld-fr — CHANGES_REQUESTED** [review]

File: elkscmd/sys_utils/init.c (hunk 2 above)
Comment (ghaerr) [reply elkscmd/sys_utils/init.c]: Nice catch, will fix.

File: elkscmd/ash/Makefile (hunk 3 above)
Comment (ghaerr) [reply elkscmd/ash/Makefile:48]: OK

File: elkscmd/rootfs_template/var/www/index.html (hunk 1 above)
Comment (ghaerr) [reply elkscmd/rootfs_template/var/www/index.html]: Currently, httpd is selected only by .config. I'll back this out.

**Review by mfld-fr — APPROVED** [review]

**Comment by mfld-fr** [comment]
Thank you @ghaerr for this big tuning step !

**Pull request merged**